source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,371 @@
+#include "extensions/stat_sinks/hystrix/hystrix.h"
+
+#include <chrono>
+#include <ctime>
+#include <iostream>
+#include <sstream>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+
+#include "server/http/admin.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+
+const uint64_t HystrixStatCache::DEFAULT_NUM_OF_BUCKETS;
+
+// Add new value to rolling window, in place of oldest one.
+void HystrixStatCache::pushNewValue(const std::string& key, uint64_t value) {
+  // Create vector if do not exist.
+  // TODO trabetti: why resize + value param didn't work without the if?
+  if (rolling_stats_map_.find(key) == rolling_stats_map_.end()) {
+    rolling_stats_map_[key].resize(window_size_, value);
+  } else {
+    rolling_stats_map_[key][current_index_] = value;
+  }
+}
+
+uint64_t HystrixStatCache::getRollingValue(absl::string_view cluster_name, absl::string_view stat) {
+  std::string key = absl::StrCat("cluster.", cluster_name, ".", stat);
+  if (rolling_stats_map_.find(key) != rolling_stats_map_.end()) {
+    // If the counter was reset, the result is negative
+    // better return 0, will be back to normal once one rolling window passes.
+    if (rolling_stats_map_[key][current_index_] <
+        rolling_stats_map_[key][(current_index_ + 1) % window_size_]) {
+      return 0;
+    } else {
+      return rolling_stats_map_[key][current_index_] -
+             rolling_stats_map_[key][(current_index_ + 1) % window_size_];
+    }
+  } else {
+    return 0;
+  }
+}
+
+void HystrixStatCache::CreateCounterNameLookupForCluster(const std::string& cluster_name) {
+  // Building lookup name map for all specific cluster values.
+  // Every call to the updateRollingWindowMap function should get the appropriate name from the map.
+  std::string cluster_name_with_prefix = absl::StrCat("cluster.", cluster_name, ".");
+  counter_name_lookup[cluster_name]["upstream_rq_5xx"] =
```

Comment:
this seems indirect. Can you capture the counter object pointer in your map, rather than capturing a long string that you'll then have to look up in updateRollingWindowMap, which seems like it might be called frequently enough to matter.